packages/expo-web-browser/android/src/main/java/expo/modules/webbrowser/WebBrowserModule.java

```diff
@@ -46,21 +53,77 @@ public String getName() {
 
   @Override
   public void setModuleRegistry(ModuleRegistry moduleRegistry) {
-    mResolver = new CustomTabsActivitiesHelper(moduleRegistry);
-    if (moduleRegistry != null) {
-      moduleRegistry.getModule(UIManager.class).registerActivityEventListener(this);
+    mCustomTabsResolver = new CustomTabsActivitiesHelper(moduleRegistry);
+    mConnectionHelper = new CustomTabsConnectionHelper(getContext());
+  }
+
+  @ExpoMethod
+  public void warmUpAsync(@Nullable String packageName, final Promise promise) {
+    try {
+      packageName = givenOfPreferredPackageName(packageName);
+      mConnectionHelper.warmUp(packageName);
+      Bundle result = new Bundle();
+      result.putString("type", "warming");
+      result.putString(SERVICE_PACKAGE_KEY, packageName);
+      promise.resolve(result);
+    } catch (NoPrefferedPackageFound ex) {
+      promise.reject(ERROR_CODE, "no_preferred_package");
+    }
+  }
+
+  @ExpoMethod
+  public void coolDownAsync(@Nullable String packageName, final Promise promise) {
+    try {
+      packageName = givenOfPreferredPackageName(packageName);
+      if (mConnectionHelper.coolDown(packageName)) {
+        Bundle result = new Bundle();
+        result.putString("type", "cooling");
+        result.putString(SERVICE_PACKAGE_KEY, packageName);
+        promise.resolve(result);
+      } else {
+        Bundle result = new Bundle();
+        result.putString("result", "Nothing to cool down");
+        result.putString(SERVICE_PACKAGE_KEY, packageName);
+        promise.resolve(result);
```

Comment:
Maybe we could resolve with an empty bundle in case of 'nothing to cool down'? 🤔